### PR TITLE
Add a new build flag to disable the pool allocator

### PR DESCRIPTION
**Pull request by rossdylan**
Calls to git_pool__malloc are proxied directly to git__malloc. Allocations are tracked using a git_vector and freed when git_pool_clear is called. I am currently using ifdef/ifndef to enable and disable the debug code. I also enable/disable a couple of the pool test cases which won't work when debug is turned on.

### CODE REVIEW

File: src/pool.h [hunk 1]
```diff
@@ -9,8 +9,13 @@
 
 #include "common.h"
 
+#ifdef GIT_DEBUG_POOL
+#include "vector.h"
+#endif
```
Comment (carlosmn) [inline src/pool.h]: This ifdef isn't necessary.

**Comment by carlosmn** [comment]
Thanks, this looks pretty straightforward. Other than the ifdef'd include, I think we could include this.

**Comment by rossdylan** [comment]
awesome, I just pushed a second commit which removes the ifdef.

File: src/pool.h [hunk 2]
```diff
@@ -98,7 +124,9 @@ extern char *git_pool_strcat(git_pool *pool, const char *a, const char *b);
 /*
  * Misc utilities
  */
+#ifndef _DEBUG_POOL
```
Comment (ethomson) [inline src/pool.h]: Should this be `GIT_DEBUG_POOL`?

Comment (rossdylan) [inline src/pool.h]: yep, looks like I typo'd that.

**Comment by ethomson** [comment]
Thanks for tackling this!

**Pull request merged**